src/alchemlyb/parsing/namd.py

```diff
@@ -25,72 +82,136 @@ def extract_u_nk(fep_file, T):
+                        print(f"namd.py: extract_u_nk: Error: l1, l2, lidws: {lambda1_at_start}, {lambda2_at_start}, {lambda_idws_at_start} changed to {lambda1}, {lambda2}, {lambda_idws}")
+                        print(f"namd.py: extract_u_nk: Error: fep_file = {fep_file}; has_idws = {has_idws}")
+                        return None
+
+                    # As we are at the end of a window, convert last window's work and times values to np arrays
+                    # (with energy unit kT since they were kcal/mol in the fepouts)
+                    win_de_arr = beta * np.asarray(win_de) # dE values
+                    win_ts_arr = np.asarray(win_ts) # timesteps
+
+                    if lambda_idws is not None:
+                        # Mimic classic DWS data
+                        # Arbitrarily match up fwd and bwd comparison energies on the same times
+                        # truncate extra samples from whichever array is longer
+                        win_de_back_arr = beta * np.asarray(win_de_back)
+                        n = min(len(win_de_back_arr), len(win_de_arr))
+
+                        tempDF = pd.DataFrame({
+                            'time': win_ts_arr[:n],
+                            'fep-lambda': np.full(n,lambda1),
+                            lambda1: 0,
+                            lambda2: win_de_arr[:n],
+                            lambda_idws: win_de_back_arr[:n]})
+
+                    else:
+                        # create dataframe of times and work values
+                        # this window's data goes in row LAMBDA1 and column LAMBDA2
+                        tempDF = pd.DataFrame({
+                            'time': win_ts_arr,
+                            'fep-lambda': np.full(len(win_de_arr), lambda1),
+                            lambda1: 0,
+                            lambda2: win_de_arr})
+
+                    # join the new window's df to existing df
+                    u_nk = pd.concat([u_nk, tempDF], sort=True)
+
+                    # reset values for next window of fepout file
+                    win_de = []
+                    win_ts = []
+                    win_de_back = []
+                    win_ts_back = []
+                    parsing = False
+
+                # append work value from 'dE' column of fepout file
+                if parsing:
+                    if l[0] == 'FepEnergy:':
+                        win_de.append(float(l[6]))
+                        win_ts.append(float(l[1]))
+                    elif l[0] == 'FepE_back:':
+                        win_de_back.append(float(l[6]))
+                        win_ts_back.append(float(l[1]))
+
+                # Turn parsing on after line 'STARTING COLLECTION OF ENSEMBLE AVERAGE'
+                if '#STARTING' in l:
+                    parsing = True
+
+    if len(win_de) != 0 or len(win_de_back) != 0:
+        print('Warning: trailing data without footer line (\"#Free energy...\"). Interrupted run?')
```

Comment:
Do not use `print()` in library code. Use a logger for *alchemlyb.parsers.NAMD* (see what the Amber parser does https://github.com/alchemistry/alchemlyb/blob/b068776354055115da6d52af24664ba3938985cc/src/alchemlyb/parsing/amber.py#L21 ) and issue a `warnings.warn` .